src/utils/fetch.ts

```diff
@@ -0,0 +1,82 @@
+export class FetchError extends Error {
+  response: {
+    status: number
+    statusText: string
+    data?: any
+  }
+
+  constructor(message: string, response: { status: number, statusText: string, data?: any }) {
+    super(message)
+    this.name = 'FetchError'
+    this.response = response
+  }
+}
+
+type FetchOptions = Omit<RequestInit, 'body'> & {
+  body?: any
+}
+
+export async function customFetch<T = any>(url: string, options: FetchOptions = {}): Promise<T> {
+  try {
+    const headers = {
+      'Content-Type': 'application/json',
+      ...options.headers,
```

Comment:
since it's needed in every call, why don't you put the `Authorization: $token` here too? It's overridable, but it spares you from manually adding it to every single `customFetch` call

Reply:
It's not needed in every call. Check login actions.

Reply:
@alvarosabu ok, anyways it could be nullable. If you think this doesn't simplify things, feel free to resolve this one